mon-pix/app/controllers/campaigns/campaign-landing-page.js

```diff
@@ -1,9 +1,43 @@
 import { action } from '@ember/object';
 import Controller from '@ember/controller';
 import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
 
 export default class CampaignLandingPageController extends Controller {
+  @service currentDomain;
+  @service intl;
+  @service locale;
   @service router;
+  @service session;
+
+  @tracked selectedLanguage = this.intl.primaryLocale;
+
+  get canDisplayLanguageSwitcher() {
+    return this.isInternationalDomain && this.isUserNotAuthenticated;
+  }
+
+  get isInternationalDomain() {
+    return !this.currentDomain.isFranceDomain;
+  }
+
+  get isUserAuthenticatedByGAR() {
+    return !!this.session.get('data.externalUser');
+  }
+
+  get isUserAuthenticatedByPix() {
+    return this.session.isAuthenticated;
+  }
+
+  get isUserNotAuthenticated() {
+    return !this.isUserAuthenticatedByPix && !this.isUserAuthenticatedByGAR;
+  }
```

Comment:
Ce code est déjà présent dans le fichier https://github.com/1024pix/pix/blob/dev/mon-pix/app/controllers/fill-in-campaign-code.js. Peut-être factoriser ces méthodes dans le service session ?

Reply:
Dans ce cas-ci, le getter `isInternationalDomain` doit aussi être factorisé.

Reply:
Peut-être qu'il y a un ticket à créer pour faire cette factorisation.